c/tskit/tables.h

```diff
@@ -92,14 +92,14 @@ typedef struct {
     tsk_flags_t flags;
     /** @brief Spatial location. The number of dimensions is defined by
      * ``location_length``. */
-    double *location;
+    const double *location;
```

Comment:
It's not so clear to me what the ``const`` is doing here. Is this documenting the fact that the memory for this object is managed by the library and should be read-only to the client?

Reply:
"managed" in the sense of read-only to the client, yes.

The main motivation for adding `const` to `double *location` was due to:
```
tsk_individual_table_get_row(const tsk_individual_table_t *self, tsk_id_t index, tsk_individual_t *row);
```
The array of locations is part of the state of `tsk_individual_table_t`. And I made that 1st parameter `const`. So if one calls `tsk_individual_table_get_row` to get a `tsk_individual_t` "row" _into_ that table, that "row" should not be able to change the state of the table.

But there is a design decision there. Looking over the code it seemed to me the intention is that these "row" objects are conceptually read-only copies of the data found in the table. In terms of literal memory they might not be entirely copied memory, but conceptually they are copies of logical state.

Reply:
The row-objects returned by ``x_table_get_row`` are definitely read-only and will remain that way, so it would be great to constify these to make this formal. Read/write row access will always be tricky because of the ragged columns, so when we want to make row-wise editing possible we should do it via a ``x_table_set_row(table, index, *params)``, where params are the same as ``x_table_add_row``. That way we can do the repacking of the ragged columns transparently (if needed).